appcelerator/titanium_mobile
# fix(android): Fixed bug where app launches without animation when launched with data intent

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-26715

**Summary:**
- This issue was introduced as of Titanium 8.0.0 due to the intent refactoring.
- The issue is that the app suddenly appears when resumed. It used to do a transition animation such as a fade-in, slide-in, etc.
- This fix will now show a fade-in effect when receiving a data intent.

**Note:**
This animation bug only happens for data intents. It does not happen when launching the app normally via the app list screen (ie: main launcher intent) or when shown via `startActivityForResult()`.

**Test:**
1. Create a classic Titanium app.
2. Set project name to "IntentTest". (This is `<name/>` in "tiapp.xml".)
3. Set project's "Application Id" to "com.appc.intent.test". (This is `<id>` in "tiapp.xml".)
4. Set up the "app.js" with the below code.
5. Build and run on Android.
6. Tap the device's "Home" button to put the app in the background.
7. Open the Mac "Terminal".
8. CD (change directory) to: `~/Library/Android/sdk/platform-tools`
9. In the terminal, enter the following...
`./adb shell am start -n com.appc.intent.test/.IntenttestActivity -a android.intent.action.VIEW -d https://www.google.com`
10. Verify that the app "fades-in" when resumed. _(This is the fix.)_

```javascript
Ti.Android.rootActivity.addEventListener("newintent", function(e) {
	label.text = "New Intent:\n" + JSON.stringify(e.intent, null, 4);
});

var window = Ti.UI.createWindow();
var scrollView = Ti.UI.createScrollView({
	width: Ti.UI.FILL,
	height: Ti.UI.FILL,
});
var label = Ti.UI.createLabel({
	text: "Launch Intent:\n" + JSON.stringify(Ti.Android.rootActivity.intent, null, 4),
	width: Ti.UI.FILL,
	height: Ti.UI.SIZE,
});
scrollView.add(label);
window.add(scrollView);
window.open();
```
